packages/router/utils.ts

```diff
@@ -959,23 +965,70 @@ export class DeferredData {
     this.subscriber?.(true);
   }
 
+  async resolveData(signal: AbortSignal) {
```

Comment:
Utility to resolve all the data for the deferred (used for fetchers and revalidations)